security/WFCORE-3873_SNI_Support.adoc

```diff
@@ -0,0 +1,76 @@
+= SNI Support
+:author:            Stuart Douglas
+:email:             sdouglas@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This aim of this issue is to provide SNI support for all SSL use cases that use SSLEngine.
+
+This will allow for users to specify match rules for incoming SNI names, and at runtime
+the appropriate SSL context will be selected for the name.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-3873[WFCORE-3873]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-726
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Affected Projects or Components
+
+* Wildfly Core
+* Undertow
+* Elytron
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+This will involve adding a new `server-sni-ssl-context` resource to the Elytron subsystem. This resource will have
```

Comment:
How is `server-sni-ssl-context` implemented? Does it "inherit" from `server-ssl-context`? Will any update of `server-ssl-context` (e.g. TLSv1.3) automatically propagate to `server-sni-ssl-context` or do we need pay attention to get to both now?

Reply:
That comment is likely more appropriate on the TLS 1.3 RFE and any resulting analysis, this is quite an old RFE now so we wouldn't retrospectively start worrying about TLS 1.3 at this stage which is a new RFE.

Reply:
My note is related to implementaiton of this feature. So I htink it belongs here. TLS 1.3 is just example. I mean in general if there is any change on `server-ssl-context`, e.g. new attribute. Will it be automatically available in `server-sni-ssl-context`. I dont see any PR with implementation 
anywhere. 

Or `server-sni-ssl-context` is copy of `server-ssl-context` with additional attributes?

Reply:
The pull request has now been merged to WildFly Core containing the subsystem changes to configure an SNI aware SSLContext.

Reply:
I looked into implementation and I have realized my concern was wrong. `server-sni-ssl-context` has only 2 attributes: DEFAULT_SSL_CONTEXT, HOST_CONTEXT_MAP. So if anything changed in `server-ssl-context` it will be in `server-sni-ssl-context` as well.